src/front/wgsl/mod.rs

```diff
@@ -3342,13 +3346,19 @@ impl Parser {
         } else if !reference.is_reference {
             return Err(Error::InvalidAssignment {
                 span: lhs_span,
-                ty: match context.expressions.get_mut(reference.handle) {
-                    crate::Expression::Swizzle { .. } => InvalidAssignmentType::Swizzle,
-                    _ => InvalidAssignmentType::Other,
+                ty: if context.named_expressions.contains_key(&reference.handle) {
+                    InvalidAssignmentType::ImmutableBinding
+                } else {
+                    match context.expressions.get_mut(reference.handle) {
+                        crate::Expression::Swizzle { .. } => InvalidAssignmentType::Swizzle,
+                        _ => InvalidAssignmentType::Other,
+                    }
```

Comment:
Let's pull this out into a temporary variable, and not have big computations in the midst of the `Error::InvalidAssignment` construction.